app/services/court_application_maat_link_creator.rb

```diff
@@ -0,0 +1,89 @@
+class CourtApplicationMaatLinkCreator < ApplicationService
+  attr_reader :laa_reference, :subject_id
+
+  def initialize(subject_id, user_name, maat_reference)
+    @laa_reference = LaaReference.new(
+      defendant_id: subject_id,
+      user_name:,
+      maat_reference: maat_reference.presence || LaaReference.generate_linking_dummy_maat_reference,
+    )
+    @subject_id = subject_id
+  end
+
+  def call
+    post_laa_references_to_common_platform
+    publish_laa_reference_to_queue unless laa_reference.dummy_maat_reference?
+    fetch_past_hearings
+
+    laa_reference.adjust_link_and_save!
+  end
+
+private
+
+  def publish_laa_reference_to_queue
+    maat_api_laa_reference = MaatApi::CourtApplicationLaaReference.new(
+      maat_reference: laa_reference.maat_reference,
+      user_name: laa_reference.user_name,
+      court_application_summary:,
+    )
+
+    Sqs::MessagePublisher.call(
+      message: MaatApi::LaaReferenceMessage.new(maat_api_laa_reference).generate,
+      queue_url: Rails.configuration.x.aws.sqs_url_link,
+      log_info: { maat_reference: laa_reference.maat_reference },
+    )
+  end
+
+  def post_laa_references_to_common_platform
+    offences.each do |offence|
+      post_laa_reference_to_common_platform(offence)
+    rescue StandardError => e
```

Comment:
I think you need to wrap the `rescue` between begin/end
something like
```
offences.each do |offence|
    begin
      post_laa_reference_to_common_platform(offence)
    rescue StandardError => e
    ...
    end
  end

Reply:
I tried that, and Rubocop gives me a `Redundant `begin` block detected.` error - turns out you don't need it.

Reply:
yes, you're right, I've just checked.
the block was necessary before Ruby 2.5,
thanks for trying!